modules/search/class-jetpack-instant-search.php

```diff
@@ -0,0 +1,367 @@
+		wp_enqueue_style( 'jetpack-instant-search', $style_path, array(), $style_version );
+	}
+
+	/**
+	 * Passes all options to the JS app.
+	 */
+	protected function inject_javascript_options() {
+		$widget_options = Jetpack_Search_Helpers::get_widgets_from_option();
+		if ( is_array( $widget_options ) ) {
+			$widget_options = end( $widget_options );
+		}
+
+		$filters = Jetpack_Search_Helpers::get_filters_from_widgets();
+		$widgets = array();
+		foreach ( $filters as $key => $filter ) {
+			if ( ! isset( $widgets[ $filter['widget_id'] ] ) ) {
+				$widgets[ $filter['widget_id'] ]['filters']   = array();
+				$widgets[ $filter['widget_id'] ]['widget_id'] = $filter['widget_id'];
+			}
+			$new_filter                                   = $filter;
+			$new_filter['filter_id']                      = $key;
+			$widgets[ $filter['widget_id'] ]['filters'][] = $new_filter;
+		}
+
+		$post_type_objs   = get_post_types( array(), 'objects' );
+		$post_type_labels = array();
+		foreach ( $post_type_objs as $key => $obj ) {
+			$post_type_labels[ $key ] = array(
+				'singular_name' => $obj->labels->singular_name,
+				'name'          => $obj->labels->name,
+			);
+		}
+
+		$prefix  = Jetpack_Search_Options::OPTION_PREFIX;
+		$options = array(
+			'overlayOptions'  => array(
+				'closeColor'      => get_option( $prefix . 'close_color', '#BD3854' ),
+				'colorTheme'      => get_option( $prefix . 'color_theme', 'light' ),
+				'enableInfScroll' => (bool) get_option( $prefix . 'inf_scroll', false ),
+				'highlightColor'  => get_option( $prefix . 'highlight_color', '#FFC' ),
+				'opacity'         => (int) get_option( $prefix . 'opacity', 97 ),
+				'showLogo'        => (bool) get_option( $prefix . 'show_logo', true ),
+				'showPoweredBy'   => (bool) get_option( $prefix . 'show_powered_by', true ),
+			),
+
+			// core config.
+			'homeUrl'         => home_url(),
+			'locale'          => str_replace( '_', '-', get_locale() ),
+			'postsPerPage'    => get_option( 'posts_per_page' ),
+			'siteId'          => Jetpack::get_option( 'id' ),
+
+			// filtering.
+			'postTypeFilters' => $widget_options['post_types'],
+			'postTypes'       => $post_type_labels,
+			'sort'            => $widget_options['sort'],
+			'widgets'         => array_values( $widgets ),
+		);
+
+		/**
+		 * Customize Instant Search Options.
+		 *
+		 * @module search
+		 *
+		 * @since 7.7.0
+		 *
+		 * @param array $options Array of parameters used in Instant Search queries.
+		 */
+		$options = apply_filters( 'jetpack_instant_search_options', $options );
+
+		// Use wp_add_inline_script instead of wp_localize_script, see https://core.trac.wordpress.org/ticket/25280.
+		wp_add_inline_script( 'jetpack-instant-search', 'var JetpackInstantSearchOptions=' . wp_json_encode( $options ) );
+	}
+
+	/**
+	 * Registers a widget sidebar for Instant Search.
+	 */
+	public function register_jetpack_instant_sidebar() {
+		$args = array(
+			'name'          => 'Jetpack Search Sidebar',
```

Comment:
This should use __( to allow translations.

__( 'Jetpack Search Sidebar', 'jetpack' )

Reply:
Fixed!